### PR TITLE
feat!: Add support for SNS topic filtering, subscriptions, and topic policy creation

### LINKED ISSUE #37: Add wrapper module generated with pre-commit
## Describe the solution you'd like.

Add wrapper module same as in SQS module

### LINKED ISSUE #38: No subscription option
No subscription option to subscribe an email to get notified

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- `create_sns_topic` has been renamed to `create`
- `policy` has been renamed to `topic_policy`
- `name_prefix` has been replaced with the combination of `name` and `use_name_prefix = true` to ensure only one value is provided
- `*_failure_feedback_role_arn`, `*_success_feedback_role_arn`, `*_success_feedback_sample_rate` variables have been replaced with a respective top level variable that contains a map to the three attributes. See before and after below for further clarification.
- Outputs have had the `sns_` prefix stripped from their names
- Add support for topic filtering, subscriptions, and topic policy creation; min version of AWS provider has been increased to support

## Motivation and Context
- Allow users to create subscriptions and policies for more advanced use cases with SNS topics
- Resolves #37
- Resolves #38

## Breaking Changes
- Yes, see the [`UPGRADE-5.0.md`](https://github.com/bryantbiggs/terraform-aws-sns/blob/feat/filtering/UPGRADE-5.0.md) guide provided

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by bryantbiggs** [comment]
opening as draft because I want to update SQS as well and make sure the two work well together for pub/sub architectures (avoid chicken vs egg or scenario when passing attributes from one to the other, etc.)

File: README.md [hunk 1]
```diff
@@ -2,34 +2,146 @@
...
+
+### FIFO Topic w/ FIFO SQS Subscription
+
 ```hcl
 module "sns_topic" {
   source  = "terraform-aws-modules/sns/aws"
-  version = "~> 3.0"
 
-  name  = "my-topic"
+  name  = "fifo"
```
Comment (antonbabenko) [inline README.md:80]:
```suggestion
  name  = "my-topic"
```

Just because the term `fifo` has a meaning in this context.

**Review by antonbabenko — APPROVED** [review]
Nice! Minor comment.

**Comment by antonbabenko** [comment]
This PR is included in version 5.0.0 :tada:

**Pull request merged**